daemon/daemon.go

```diff
@@ -482,6 +499,10 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 		return fmt.Errorf("internal error: no Overlord")
 	}
 
+	if d.cancel != nil {
+		d.cancel()
```

Comment:
this probably needs an explicit test in daemon_test.go

Reply:
I"ve added couple more tests that verify context cancelation and propagation, and one which hijacks d.cancel